### PR TITLE
Introduce Omni lock into Godwoken and rm clerkb (POA)

### PR DESCRIPTION
**Pull request by Flouse**
- replace https://github.com/RetricSu/godwoken-kicker/pull/171
- omni_lock has been added into ghcr.io/flouse/godwoken-prebuilds:v1.0.x-omni-lock-202203051513
  see: https://github.com/Flouse/godwoken-docker-prebuilds/blob/v1.0.x-omni-lock/Makefile#L13
- test Kicker while sending PR to compatibility-changes branch
   https://github.com/RetricSu/godwoken-kicker/compare/compatibility-changes...omni-lock#diff-487ff6372ad077cb90406a903f435752a206d27afd9553d9ba1212412410708fR7

- [x] fix Makefile:66: recipe for target 'install' failed
        make[1]: *** [install] Error 1
        Makefile:117: recipe for target 'workspace/bin/godwoken' failed
        make: *** [workspace/bin/godwoken] Error 2

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -325,7 +325,7 @@ enter-kafka:
 
 ########### manual-build-mode #############
 ### rebuild components's scripts and bin all in one
-rebuild-scripts: rebuild-gw-scripts-and-bin rebuild-polyjuice-bin rebuild-poa-scripts
+rebuild-scripts: rebuild-gw-scripts-and-bin rebuild-polyjuice-bin
```
Comment (RetricSu) [inline Makefile:328]:
```suggestion
rebuild-scripts: rebuild-gw-scripts-and-bin rebuild-polyjuice-bin rebuild-omni-lock
```

File: docker/.build.mode.env [hunk 2]
```diff
@@ -6,10 +6,11 @@ MANUAL_BUILD_SCRIPTS=false
 MANUAL_BUILD_POLYJUICE=false
 MANUAL_BUILD_CLERKB=false
```
Comment (RetricSu) [inline docker/.build.mode.env:7]:
```suggestion
```
remove this poa env?

File: .gitignore [hunk 3]
```diff
@@ -17,7 +17,7 @@ cargo-cache-data/
 quick-mode/
 
 config/scripts/release/*
-!config/scripts/release/poa
+
 !config/scripts/release/state
```
Comment (RetricSu) [inline .gitignore:21]:
```suggestion
```

Comment (RetricSu) [reply .gitignore:21]: state is also poa, can be removed too

File: docker/.build.mode.env [hunk 4]
```diff
@@ -34,6 +34,8 @@ POLYJUICE_GIT_URL=https://github.com/nervosnetwork/godwoken-polyjuice.git
 POLYJUICE_GIT_CHECKOUT=eth-addr-mapping-error
 CLERKB_GIT_URL=https://github.com/nervosnetwork/clerkb.git
 CLERKB_GIT_CHECKOUT=v0.4.0
```
Comment (keroro520) [inline docker/.build.mode.env:36]: clerkb configurations should be removed now

File: docker/layer2/entrypoint.sh [hunk 5]
```diff
@@ -123,10 +123,10 @@ echo 'start deploying godwoken scripts, this might takes a little bit of time, p
 deployGodwokenScripts 5 $POLYMAN_RPC "/code/workspace/deploy/scripts-deploy.json" "/code/workspace/deploy/scripts-deploy-result.json" 
 
 # deploy genesis block
-$GW_TOOLS_BIN deploy-genesis --ckb-rpc ${CKB_RPC} --scripts-deployment-path deploy/scripts-deploy-result.json -r deploy/rollup-config.json -o deploy/genesis-deploy-result.json -k ${PRIVKEY}
+$GW_TOOLS_BIN deploy-genesis --ckb-rpc ${CKB_RPC} --scripts-deployment-path deploy/scripts-deploy-result.json -r deploy/rollup-config.json --omni-lock-config-path deploy/scripts-deploy-result.json -o deploy/genesis-deploy-result.json -k ${PRIVKEY}
```
Comment (keroro520) [inline docker/layer2/entrypoint.sh:126]: `--omni-lock-config-path deploy/scripts-deploy-result.json` ? Not omni config?

**Review by keroro520 — CHANGES_REQUESTED** [review]

File: docker/layer2/entrypoint.sh (hunk 5 above)
Comment (Flouse) [reply docker/layer2/entrypoint.sh:126]: The omni config supposed to be in `deploy/scripts-deploy-result.json`.

see: https://github.com/RetricSu/godwoken-kicker/pull/172/files#diff-6f62da3aed030fbfd698e10fde099189b986aeb385d34a3ffac8365408a480feR19

**Comment by Flouse** [comment]
## Test `make start` passed in CI
https://github.com/RetricSu/godwoken-kicker/runs/5437886319?check_suite_focus=true#step:4:644

@RetricSu @keroro520 @zeroqn 
Please review again, thx.

File: docker/layer2/entrypoint.sh (hunk 5 above)
Comment (keroro520) [reply docker/layer2/entrypoint.sh]: I am not sure. I just notice that the Rust definition [`OmniLockConfig`](https://github.com/zeroqn/godwoken/blob/c18807b5cfaa961c230e15e3a381570c324db6f8/crates/tools/src/types.rs#L70-L74) is not a `Script`.

Comment (keroro520) [reply docker/layer2/entrypoint.sh]: @zeroqn Could you confirm it?

Comment (zeroqn) [reply docker/layer2/entrypoint.sh]: Yes

**Pull request merged**